### DISCUSSION TITLE
Strange BRouter route with unpaved road

### DISCUSSION THREAD

**Discussion by tweaks90** [devemux86/cruiser#504] [Q&A] [answered]
Hi

I have been using cruiser (and previously kurviger) for years and with great passion. I have only recently started to use BRouter (offline) though as it is perfectly integrated with Cruiser nowadays.

I do occasionally get strange results with BRouter, which - most likely - are related to my settings.

I attach screen-shots of the routes to better illustrate. The 1st screen-shot displays the route as it "should" be. This is the result when using OpenRouteService. Also, I have driven this route a couple times and actually know it quite well. It does have sections with gravel roads.

![image](https://github.com/devemux86/cruiser/assets/75031892/e0c52c36-03c7-4f69-9bc4-abdd6d7166af)

The 2nd screen-shot shows how brouter calculates the route (default settings for fast and curvy):
![image](https://github.com/devemux86/cruiser/assets/75031892/f156ce91-216e-4100-9df8-f21ad22921e8)

I can largely influence the route by changing the speed to. With a setting of 75 km/h for instance I get:
![image](https://github.com/devemux86/cruiser/assets/75031892/b9ef1425-3e50-4980-870b-15cd3738625b)

Finally, with a setting of 35km/h it does look the same as the one obtained with OpenRouteService.

I have also played with other settings (e.g. explicitly enable gravel roads.) It does have some impact, but always takes a - more or less - large detour.

I noted that brouter also comes up with detours in other areas, so its not limited to this particular region. I have already experimented a bit with a speed setting of 35km/h (was on the road a few times) but this does have other drawbacks. Most notably, it will almost *always* lead you to any possible side road it will find (some of which not even publicly accessible) no matter what. Not ideal but I guess that's the idea of the speed option. So changing the speed might not be an option.

--> Any hints as to where the big detours come from?

What alternatives to BRouter could you recommend. I am soon going on a larger tour in territory totally unknown to me, so need to rely more on the routing engine.

Thanks,
Tom

**Comment by devemux86** [comment]
Hi @tweaks90 and thanks for the very interesting route example!

BRouter is the best **offline** routing service with unlimited ways to customize the routing.

Examining the road tags, the problem seems to be this road with "surface=unpaved":
https://www.openstreetmap.org/way/27666510

Like OpenRouteService, also GraphHopper, OSRM, Valhalla on OpenStreetMap website use this road!

BRouter seems to be more strict, even with allow unpaved, it sets them with low priority for vehicles,
so generally they tend to not use them if there are better road surfaces.

_But if we change the unpaved rule inside the profile files, then BRouter can use more such roads too._

So probably we need to make some changes about the use of unpaved roads in Cruiser profile.
(make unpaved more usable if their option is set to allow their use)

**Comment by devemux86** [comment]
I found where we need to make the change for the unpaved roads.

Then BRouter calculates a similar route like the other routers.
Image from [BRouter website](https://brouter.de/brouter-web/#map=12/46.5350/15.3589/standard&lonlats=15.356483,46.583406;15.34112,46.466005) testing the modified profile:

![BRouter](https://github.com/devemux86/cruiser/assets/3484020/d2131fa8-a98b-482e-b493-1fa72cdc69bc)

**Reply by devemux86** [reply]
Cruiser driving profile with the unpaved improvement is here:
- https://github.com/devemux86/cruiser/discussions/272

**Reply by bios4** [reply]
> I found where we need to make the change for the unpaved roads.

Could you please share WHAT EXACTLY was the change to make the result look more like expected?
The info could be useful to "learn" some more details about what settings affect the BRouter engine in which way...

Thanks in advance!

**Reply by devemux86** [reply]
Line 189 in Cruiser driving profile, change to `switch avoid_unpaved 1 10`.
(was `switch avoid_unpaved 1 2`)

**Reply by bios4** [reply]
Thanks.

And yes, this can be seen when comparing the old profile with the new one...
What I meant is "what does this mean in understandable words"? 😁

Sorry if my question was not clear enough...

**Reply by devemux86** [reply]
With `2`, BRouter sets very low priority to unpaved roads for motor vehicles, even if their avoid option is off.

But if their option allows them, then user wants to use them, it should be similar to all other vehicle routers.
Value `10` seems a nice improvement based on the current tests.

**Reply by bios4** [reply]
Thanks for explaining (I'm sure I'm NOT the only one who has NO idea about that BRouter stuff)!

**Reply by devemux86** [reply]
I always study the BRouter rules between Cruiser development! :slightly_smiling_face: 

The best thing about BRouter is that we can change the routing so much with some "simple" profile changes.

**Comment by tweaks90** [comment]
Count me in on having absolutely NO idea about BRouter ;-)

Thanks for the analysis @devemux86. Do i need to manually install/edit the profile directly on my mobile or would this automagically be installed with the next (beta) release?

Thanks,
Tom

**Reply by devemux86** [reply]
@tweaks90 the improved profile will be included automatically in the next Cruiser update.

**Reply by devemux86** [reply]
@tweaks90 

The improved Cruiser driving profile is now available in the latest:
- https://github.com/devemux86/cruiser/discussions/482

**Reply by tweaks90** [reply]
Thanks for the quick fix. This works proper now.

**Comment by devemux86** [comment, marked as answer]
Improved unpaved roads with BRouter is available in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
